### DISCUSSION TITLE
Prevent enter route

### DISCUSSION THREAD

**Discussion by hugo-cardoso** [TanStack/router#2559] [Q&A]
I would like, given a business rule applied in `beforeLoad`, navigation to be cancelled.

```tsx
export const Route = createFileRoute("/my-page")({
  component: () => <Page />,
  beforeLoad() {
    const isAllowed = false;

    if (!isAllowed) {
      // prevent entering the page
    }
  }
});
```

Is this possible?

References:

https://router.vuejs.org/guide/advanced/navigation-guards.html#Per-Route-Guard
https://router.vuejs.org/guide/advanced/navigation-guards.html#Global-Before-Guards

**Comment by schiller-manuel** [comment]
you could throw a redirect

**Reply by hugo-cardoso** [reply]
For my scenario I would indeed need to cancel the navigation and keep the user on the route he is already on.

https://router.vuejs.org/guide/advanced/navigation-guards.html#Global-Before-Guards

**Reply by schiller-manuel** [reply]
and you would like to evaluate this in `beforeLoad` or rather in such a global guard as you linked to?

**Reply by hugo-cardoso** [reply]
The 2 options are very useful. But for my specific scenario I think the ideal would be a local guard.

https://router.vuejs.org/guide/advanced/navigation-guards.html#Per-Route-Guard

**Reply by schiller-manuel** [reply]
so I can imagine that you could throw a `cancelNavigation()` in `beforeLoad`

**Reply by hugo-cardoso** [reply]
It would be great to be able to do that!

**Reply by schiller-manuel** [reply]
hm no this probably won't work. `beforeLoad` is executed after the location has already been committed.

**Reply by hugo-cardoso** [reply]
This behavior is curious. The documentation says that `beforeLoad` runs before the route is loaded.

Does this mean that at this moment he has already destroyed the previous route?

**Reply by hugo-cardoso** [reply]
Hi! Is there any implementation planned to support this or does it not make sense at the moment?

**Reply by schiller-manuel** [reply]
nothing planned at the moment, but could be further investigated (not a priority right now though)
